packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -319,19 +319,20 @@ class IconButton extends StatelessWidget {
       minHeight: _kMinButtonSize,
     );
     final BoxConstraints adjustedConstraints = effectiveVisualDensity.effectiveConstraints(unadjustedConstraints);
+    final double effectiveIconSize = iconSize ?? IconTheme.of(context).size ?? 24.0;
```

Comment:
It looks like above we have already captured

```dart
final ThemeData theme = Theme.of(context);
```

Is `IconTheme.of(context)` the same as `theme.iconData`? If so we can simplify this, but I do not know if this is true. cc @darrenaustin and @HansMuller  can you advise?

Reply:
Actually can you add a test case for an icon theme that is part of the ThemeData? That would tell us pretty easily. :)

Reply:
`IconTheme.of(context)` covers any `IconTheme` as its ancestor too. `theme.iconData` would not cover it.

<details>
<summary>code example</summary>

```dart
// when final double effectiveIconSize = iconSize ?? theme.iconTheme.size ?? 24.0;
IconTheme(
  data: const IconThemeData(size: 10),
  child: IconButton(
  onPressed: () {},
  icon: const Icon(Icons.link),
  // iconSize: 24.0
),

// when final double effectiveIconSize = iconSize ?? IconTheme.of(context).size ?? 24.0;
IconTheme(
  data: const IconThemeData(size: 10),
  child: IconButton(
  onPressed: () {},
  icon: const Icon(Icons.link),
  // iconSize: 10.0
),
```
</details>

Reply:
Right, IconTheme.of() returns a value that's guaranteed to have non-null properties. It's up to the Material library widgets to create/configure IconTheme's using defaults from the Material theme.